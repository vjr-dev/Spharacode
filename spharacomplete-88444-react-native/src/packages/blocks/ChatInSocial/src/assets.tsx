// Customizable Area Start
export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");
export const image_back = require("../assets/image_back.png");
export const LOGO = require("../assets/LOGO.png");
export const Pin = require("../assets/Pin.png");
export const Delete = require("../assets/Delete.png");
export const Archive = require("../assets/Archive.png");
export const More = require("../assets/More.png");
export const menu = require("../assets/Menu.png");
export const search = require("../assets/search.png");
export const check = require("../assets/check.png");
// Customizable Area End