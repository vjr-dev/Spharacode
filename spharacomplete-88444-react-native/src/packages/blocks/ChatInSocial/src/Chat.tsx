// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import {
  ActivityIndicator,
  Dimensions,
  FlatList,
  Image,
  ImageBackground,
  Modal,
  SafeAreaView,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from "react-native";

import Swipeout from "react-native-swipeout";
import { COLORS } from "../../../framework/src/Globals";
import * as IMAGE from "./assets";
import ChatController from "./ChatController";
import { Styles } from "./ChatStyle";

const windowWidth = Dimensions.get("window").width;
const windowHeight = Dimensions.get("window").height;

export default class Chat extends ChatController {
  label = (label: string) => {
    if (label === "pin") {
      return "Pin";
    } else if (label === "delete") {
      return "Delete";
    } else if (label === "archive") {
      return "Archive";
    } else {
      return "More";
    }
  };
  imageSource = (label: string) => {
    if (label === "pin") {
      return IMAGE.Pin;
    } else if (label === "delete") {
      return IMAGE.Delete;
    } else if (label === "archive") {
      return IMAGE.Archive;
    } else {
      return IMAGE.menu;
    }
  };
  component(item: any, label: string) {
    return (
      <TouchableOpacity
        testID="componentButton"
        onPress={() => {
          if (label === "pin") {
            this.pinClick(item);
          } else if (label === "delete") {
            this.DeleteClick(item);
          } else if (label === "archive") {
            this.ArchiveClick(item);
          } else {
            this.MoreClick(item);
          }
        }}
        style={Styles.swipMenuButton}
      >
        <Image
          resizeMode="contain"
          source={this.imageSource(label)}
          style={Styles.swipeMenuImage}
        />
        <Text style={Styles.swipeMenuText}>{this.label(label)}</Text>
      </TouchableOpacity>
    );
  }
  swipeoutOne(item, index) {
    return (
      <Swipeout
        close={this.state.close}
        autoClose={true}
        style={{ marginTop: index == 0 ? 10 : 13 }}
        backgroundColor="trancperant"
        left={[
          {
            component: this.component(item, "pin"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "delete"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "archive"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "more"),
            backgroundColor: COLORS.orangelight,
          },
        ]}
      >
        <TouchableOpacity
          onPress={() => this.onclick()}
          style={Styles.chatPersonList}
        >
          <View style={[Styles.chatPersonImageView, { width: "20%" }]}>
            <TouchableOpacity
              testID="chatItemButton"
              onPress={() => this.onclick()}
              style={Styles.chatPerson}
            >
              {item?.avatar ? (
                <Image
                  resizeMode="cover"
                  style={Styles.chatPersonImage}
                  source={{ uri: item.avatar }}
                />
              ) : (
                <View style={Styles.contactText}>
                  <Text
                    style={{
                      textTransform: "uppercase",
                      color: "black",
                      fontWeight: "bold",
                      fontSize: 18,
                    }}
                  >
                    {item?.name.charAt(0)}
                  </Text>
                </View>
              )}
            </TouchableOpacity>
          </View>
          <TouchableOpacity
            testID="conversationButton"
            onPress={() =>
              this.props.navigation.navigate("ConversationScreen", {
                chatNumber: item.uid,
                mainConversationId: item.conversationId,
                userName: item.name,
                userAvtar: item.avatar,
                userType: 1,
              })
            }
            style={Styles.chatPersonNameView}
          >
            <Text style={Styles.chatPersonName}>{item?.name}</Text>
            <Text style={{ color: COLORS.lightwhite }}>
              {item?.lastMessage}
            </Text>
          </TouchableOpacity>
          <View style={[Styles.chatPersonImageView, { width: "15%" }]}>
            <Text
              style={{
                color: item?.Count == 0 ? COLORS.lightwhite : COLORS.white,
              }}
            >
              {item?.Time}
            </Text>

            {item?.unReadCounts == 0 || isNaN(item?.unReadCounts) ? null : (
              <View style={Styles.chatUnreadMessage}>
                <Text>{item.unReadCounts}</Text>
              </View>
            )}
          </View>
        </TouchableOpacity>
      </Swipeout>
    );
  }

  swipeoutTwo(item, index) {
    return (
      <Swipeout
        close={this.state.close}
        autoClose={true}
        style={{ marginTop: index == 0 ? 10 : 13 }}
        backgroundColor="trancperant"
        left={[
          {
            component: this.component(item, "pin"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "delete"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "archive"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "more"),
            backgroundColor: COLORS.orangelight,
          },
        ]}
      >
        <TouchableOpacity
          // onPress={() => this.onclick()}
          style={Styles.chatPersonList}
        >
          <View style={[Styles.chatPersonImageView, { width: "20%" }]}>
            <TouchableOpacity
              testID="chatItemButton"
              onPress={() => this.onclick()}
              style={Styles.chatPerson}
            >
              {item?.icon ? (
                <Image
                  resizeMode="contain"
                  style={Styles.chatPersonImage}
                  source={{ uri: item.icon }}
                />
              ) : (
                <View style={Styles.contactText}>
                  <Text
                    style={{
                      textTransform: "uppercase",
                      color: "black",
                      fontWeight: "bold",
                      fontSize: 18,
                    }}
                  ></Text>
                </View>
              )}
            </TouchableOpacity>
          </View>
          <TouchableOpacity
            testID="conversationButton"
            onPress={() =>
              this.props.navigation.navigate("ConversationScreen", {
                chatNumber: item.guid,
                mainConversationId: item.conversationId,
                userName: item.name,
                userAvtar: item.icon,
                userType: 2,
              })
            }
            style={Styles.chatPersonNameView}
          >
            <Text style={Styles.chatPersonName}>{item?.name}</Text>
            <Text style={{ color: COLORS.lightwhite }}>
              {item?.lastMessage}
            </Text>
          </TouchableOpacity>
          <View style={[Styles.chatPersonImageView, { width: "15%" }]}>
            <Text
              style={{
                color: item?.Count == 0 ? COLORS.lightwhite : COLORS.white,
              }}
            >
              {item?.Time}
            </Text>
          </View>
        </TouchableOpacity>
      </Swipeout>
    );
  }
  addFlatList() {
    return (
      <>
        <FlatList
          data={
            this.state.Section == 1
              ? this.state.Chat_Data
              : this.state.Group_Data
          }
          testID="conversationList"
          style={{ width: windowWidth, flexGrow: 1 }}
          contentContainerStyle={{ paddingBottom: 500 }}
          showsVerticalScrollIndicator={false}
          keyExtractor={(index): any => index}
          renderItem={({ item, index }: any) => {
            return this.state.Section == 1
              ? this.swipeoutOne(item, index)
              : this.swipeoutTwo(item, index);
          }}
        />
      </>
    );
  }

  ConditionOne() {
    return (
      <>
        {this.state.Loader ? (
          <View style={{ flex: 1 }}>
            <Modal visible={this.state.Loader} transparent={true}>
              <View
                style={{
                  height: windowHeight,
                  width: windowWidth,
                  alignItems: "center",
                  justifyContent: "center",
                }}
              >
                <View
                  style={{
                    height: 100,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    justifyContent: "center",
                    alignItems: "center",
                  }}
                >
                  <ActivityIndicator
                    animating={true}
                    size={"large"}
                    color="#f07233"
                  />
                </View>
              </View>
            </Modal>
          </View>
        ) : (
          <View style={{ flexGrow: 1 }}>
            {this.state.totalUnReadCount > 0 ? (
              <Text style={Styles.unreadMessage}>
                You have {this.state.totalUnReadCount} unread message
              </Text>
            ) : null}
            {this.addFlatList()}
          </View>
        )}

        {this.ConditionTwo()}
      </>
    );
  }

  ConditionTwo() {
    return (
      <>
        {this.state.modal1Visible || this.state.modal2Visible ? (
          <View style={Styles.shadowView} />
        ) : null}
        {this.state.modal1Visible ? (
          <View style={{ flex: 1 }}>
            <View style={Styles.modalView}>
              <View style={Styles.menuView}>
                <TouchableOpacity>
                  <Text style={Styles.menuText}>New Chat</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Edit Chats</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>New Broadcast</Text>
                </TouchableOpacity>
              </View>
              <TouchableOpacity
                testID="cancelButton"
                onPress={() => this.setState({ modal1Visible: false })}
                style={[Styles.cancelButton, { marginTop: 10 }]}
              >
                <Text style={Styles.cancelText}>CANCEL</Text>
              </TouchableOpacity>
            </View>
          </View>
        ) : null}
        {this.state.modal2Visible ? (
          <View style={{ flex: 1 }}>
            <View style={Styles.modalView}>
              <View style={Styles.menuView}>
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Mute</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Mark as unread</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Contact info</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Clear Chat</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={[Styles.menuText, { color: COLORS.red }]}>
                    Delete Chat
                  </Text>
                </TouchableOpacity>
              </View>
              <TouchableOpacity
                testID="secondModalCancelButton"
                onPress={() => this.setState({ modal2Visible: false })}
                style={[Styles.cancelButton, { marginTop: 15 }]}
              >
                <Text style={Styles.cancelText}>CANCEL</Text>
              </TouchableOpacity>
            </View>
          </View>
        ) : null}
      </>
    );
  }
  render() {
    return (
      <ImageBackground source={IMAGE.back1} style={Styles.container}>
        <ImageBackground source={IMAGE.back2} style={Styles.container}>
          <SafeAreaView>
            <View
              style={[
                Styles.headerView,
                { elevation: this.state.modal1Visible ? 0 : 10 },
              ]}
            >
              <View style={{ width: "16%" }} />
              <View style={Styles.headerCenterView}>
                <TouchableOpacity
                  testID="switchbtnID"
                  onPress={() => this.checkStateUpdate(1)}
                  style={[
                    Styles.switchbutton,
                    {
                      backgroundColor:
                        this.state.Section == 1
                          ? COLORS.orangelight
                          : "transparent",
                    },
                  ]}
                >
                  <Text
                    style={[
                      Styles.switchButtonText,
                      {
                        color:
                          this.state.Section == 1
                            ? COLORS.black
                            : COLORS.lightwhite,
                      },
                    ]}
                  >
                    Chats
                  </Text>
                </TouchableOpacity>
                <TouchableOpacity
                  testID="groupButton"
                  onPress={() => this.checkStateUpdate(2)}
                  style={[
                    Styles.switchbutton,
                    {
                      backgroundColor:
                        this.state.Section == 2
                          ? COLORS.orangelight
                          : "transparent",
                      marginRight: 6,
                    },
                  ]}
                >
                  <Text
                    style={[
                      Styles.switchButtonText,
                      {
                        color:
                          this.state.Section == 2
                            ? COLORS.black
                            : COLORS.lightwhite,
                      },
                    ]}
                  >
                    Groups
                  </Text>
                </TouchableOpacity>
              </View>
              <TouchableOpacity
                testID="backHomeButton"
                onPress={() => this.GoBackToHomeScreen()}
                style={Styles.headerEndView}
              >
                <Image source={IMAGE.LOGO} style={Styles.headerLogo} />
              </TouchableOpacity>
            </View>

            <View style={Styles.searchContainer}>
              <View style={Styles.searchView}>
                <Image source={IMAGE.search} style={Styles.searchImage} />
                <TextInput
                  testID="searchTextInput"
                  placeholder="Search"
                  onChangeText={(e) => this.onSearch(e)}
                  placeholderTextColor="white"
                  style={Styles.searchInput}
                />
              </View>
              <View style={Styles.searchMenuView}>
                <TouchableOpacity
                  testID="menuButton"
                  onPress={() => this.props.navigation.navigate("NewChat")}
                  style={Styles.searchMenuButton}
                >
                  <Image
                    resizeMode="contain"
                    source={IMAGE.menu}
                    style={Styles.serachMenuImage}
                  />
                </TouchableOpacity>
              </View>
            </View>
            {this.ConditionOne()}
          </SafeAreaView>
        </ImageBackground>
      </ImageBackground>
    );
  }
}
// Customizable Area End
