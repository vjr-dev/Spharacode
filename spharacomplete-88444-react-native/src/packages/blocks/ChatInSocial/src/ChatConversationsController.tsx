// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { BlockComponent } from "framework/src/BlockComponent";
import { IBlock } from "framework/src/IBlock";
import { runEngine } from "framework/src/RunEngine";
import { CometChat } from "@cometchat-pro/react-native-chat";
export const configJSON = require("./config");
export interface Props {
  navigation: any;
  id: string;
}
interface S {
  screenFlag:string,
  searchText: string,
  isSwipeoutClose: boolean,
  conversationList: any,
  isLoading: boolean,
  close: boolean,
  isFlatListFooterLoading: boolean,
  isVisibleUserSelectionScreen: boolean,
  unReadCount: number
}
interface SS {
  id: any;
}

export default class ChatConversationsController extends BlockComponent<Props, S, SS> {
  userConversationsRequest: any;
  groupConversationsRequest: any;
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.state = {
        screenFlag: props.screenFlag,
        searchText: '',
        isSwipeoutClose:false,
        conversationList:[],
        isLoading: true,
        close:false,
        isFlatListFooterLoading: false,
        isVisibleUserSelectionScreen: false,
        unReadCount:0
     

    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
  static getDerivedStateFromProps(props, state) {
    if(props.screenFlag !== state.screenFlag){
     
      return{
        screenFlag: props.screenFlag,
        conversationList:[]
      }
    }
  }
  componentDidUpdate(prevProps){
    if(prevProps.screenFlag !== this.state.screenFlag){
      this.setState({isLoading:true})
      this.getAllUnReadMessageCount();
      this.resetConversationRequest();
      this.getConversations();
    }
  }
   componentDidMount(): Promise<void> {
    this.getAllUnReadMessageCount();
      this.resetConversationRequest();
      this.getConversations();
  }
  resetConversationRequest(){
    this.setState({conversationList:[]})
    if(this.state.screenFlag === 'chats'){
        this.userConversationsRequest = new CometChat.ConversationsRequestBuilder()
        .setLimit(30)
        .setConversationType('user')
        //.setTags(['pin'])
        .withTags(true)
        .build();
     
    }else{
      this.groupConversationsRequest = new CometChat.ConversationsRequestBuilder()
      .setLimit(30)
     // .setTags(['pin'])
     .withTags(true)
      .setConversationType('group')
      .build();
    }
  }
  getConversations(){
    if(this.state.screenFlag === 'chats'){
      this.fetchUserConversation();
    }else{
      this.fetchGroupConversation();
    }
       
  }
  fetchUserConversation(){
    this.userConversationsRequest.fetchNext().then(
      (conversationList) => {
        let tempConversationList = this.state.conversationList;
        tempConversationList = [...this.state.conversationList,...conversationList];
        if(this.state.searchText && conversationList?.length == 30){
          this.fetchUserConversation();
        }
        this.setState({conversationList:tempConversationList, isLoading:false, isFlatListFooterLoading:false})
      },
      (error) => {
      
      }
    );
  }
  async fetchGroupConversation(){
    this.groupConversationsRequest.fetchNext().then(
      async (conversationList)  => {
        let tempConversationList = this.state.conversationList;
        tempConversationList = [...this.state.conversationList,...conversationList];
        if(this.state.searchText && conversationList?.length == 30){
          this.fetchGroupConversation();
        }
        this.setState({conversationList:tempConversationList, isLoading:false, isFlatListFooterLoading:false})
      },
      (error) => {
      }
    );
  }
  getGroupUnReadCount(){
    CometChat.getUnreadMessageCountForAllGroups().then((res:any)=>{
      let TotalUnReadCount = 0;
      for (const key in res) {
        if (res.hasOwnProperty(key)) {
        TotalUnReadCount = TotalUnReadCount + res[key];
        }
      }
      this.setState({unReadCount: TotalUnReadCount})
    })
  }
  getUserUnReadCount(){
    CometChat.getUnreadMessageCountForAllUsers().then((res:any)=>{
      let TotalUnReadCount = 0;
      for (const key in res) {
        if (res.hasOwnProperty(key)) {
        TotalUnReadCount = TotalUnReadCount + res[key];
        }
      }
      this.setState({unReadCount: TotalUnReadCount})
    })
  }
  getAllUnReadMessageCount(){
    if(this.state.screenFlag === 'groups'){
     this.getGroupUnReadCount();
    }else{
      this.getUserUnReadCount();
    }
  }
  onSearch(text: string){
    if(this.state.searchText === ''){
      this.getConversations();
    }
    this.setState({searchText: text})
  }
  
  setTagsOnCoversation(conversationID:string, conversationType: string,tags: Array<string>){
  CometChat.tagConversation(conversationID, conversationType, tags).then(
        conversation => {
            console.log('conversation', conversation);
        }, error => {
            console.log('error while fetching a conversation', error);
        }
    );
  }
  pinClick(item: any) {
    if(item?.conversationType=== 'group'){
      this.setTagsOnCoversation(item?.conversationWith?.guid,'group',['pin'])
    }else{
      this.setTagsOnCoversation(item?.conversationWith?.uid,'user',['pin'])
    }
  }
  DeleteClick(item: any) {
    this.setState({isLoading: true});
    let id;
    let type;
    if(item?.conversationType === 'group'){
      id = item?.conversationWith?.guid;
      type = 'group'

    }else{
      id = item?.conversationWith?.uid;
      type = 'user'
    }
    CometChat.deleteConversation(id, type).then(
      deletedConversation => {
          console.log(deletedConversation);
          this.setState({conversationList:[]})
          this.resetConversationRequest();
          this.getConversations();
      }, error => {
        this.setState({isLoading: false});
          console.log('error while deleting a conversation', error);
      }
  );
    
  }
  ArchiveClick(item: any) {
   
  }
  MoreClick(item: any) {
    
  }
  onPressItem(item: any){
    if(item?.conversationType === 'group'){
      this.props.navigation.navigate("ConversationScreen", {
        chatNumber: item?.conversationWith?.guid,
        mainConversationId: item?.conversationId,
        userName: item?.conversationWith?.name,
        userAvtar: item?.conversationWith?.avatar,
        userType: 2,
    })
    }else{
      this.props.navigation.navigate("ConversationScreen", {
        chatNumber: item?.conversationWith?.uid,
        mainConversationId: item?.conversationId,
        userName: item?.conversationWith.name,
        userAvtar: item?.conversationWith?.avatar,
        userType: 1,
    })
    }
  }
  onCreate(){
    if(this.state.screenFlag === 'groups'){
      this.props.navigation.navigate('UsersSelection',{title:"Select Participants ",multipleSelection:true});
    }else{
      this.props.navigation.navigate('UsersSelection',{title:"Select User",multipleSelection:false});
    }
  }
  onRefresh(){
    this.setState({conversationList:[],isLoading:true})
    this.getAllUnReadMessageCount();
    this.resetConversationRequest();
    this.getConversations();
  }
  endReached(){
    if(this.state.conversationList?.length >= 30 && !this.state.searchText){
      this.setState({isFlatListFooterLoading: true})
      this.getConversations()
    }
  }
  onBackPressOfUserSelection(){
    this.setState({isVisibleUserSelectionScreen:false})
  }
  onUserSelect(users:any){
    this.setState({isVisibleUserSelectionScreen:false})
  }
}
  
// Customizable Area End
