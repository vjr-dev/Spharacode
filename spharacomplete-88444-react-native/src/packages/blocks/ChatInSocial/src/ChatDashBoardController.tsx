// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { BlockComponent } from "framework/src/BlockComponent";
import { IBlock } from "framework/src/IBlock";
import { runEngine } from "framework/src/RunEngine";
import { GotoHomePage } from "../../../components/src/Navigation/NavigationFunctions";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { loginOnChat } from "../../../components/src/CometChatCommonFunctions";
import { Alert } from "react-native";

export const configJSON = require("./config");
export interface Props {
  navigation: any;
  id: string;
}
interface S {
  screenFlag:string,
  isUserLoggedIn: boolean
}
interface SS {
  id: any;
}

export default class ChatDashBoardController extends BlockComponent<Props, S, SS> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.state = {
        screenFlag:'chats',
        isUserLoggedIn: false
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
  componentDidMount(): Promise<void> {
    this.getInitialData();
  }
  async getInitialData(){
    const tempUserData = await AsyncStorage.getItem("User_Data");
    const userData = JSON.parse(tempUserData);
    loginOnChat(userData.data.attributes?.full_phone_number).then((response:any)=>{
      this.setState({isUserLoggedIn: true})
    }).catch((error: any)=>{
      Alert.alert("","Something went wrong, please try again")
    })
  }
	onSwitch(screen:string){
		this.setState({screenFlag: screen})
	}
	onPressLogo(){
		GotoHomePage(this.props);
	}
}
// Customizable Area End
