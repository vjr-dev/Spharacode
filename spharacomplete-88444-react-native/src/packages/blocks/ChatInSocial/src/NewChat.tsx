// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import {
  Dimensions,
  FlatList,
  Image,
  ImageBackground,
  SafeAreaView,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from "react-native";
import { COLORS } from "../../../framework/src/Globals";
import * as IMAGE from "./assets";
import NewChatController, { Props } from "./NewChatController";
import { Styles } from "./NewChatStyle";

const windowWidth = Dimensions.get("window").width;

export default class NewChat extends NewChatController {
  constructor(props: Props) {
    super(props);
  }

  renderItem = ({ item, index }: any) => (
    <View style={{ marginTop: 18, justifyContent: "center" }}>
      <View style={{ marginHorizontal: 12, flexDirection: "row" }}>
        <View style={{ width: "18%" }}>
          <TouchableOpacity>
            {item.avatar ? (
              <Image
                resizeMode="contain"
                style={Styles.chatPersonImage}
                source={{ uri: item.avatar }}
              ></Image>
            ) : (
              <View style={Styles.contactText}>
                <Text
                  style={{
                    textTransform: "uppercase",
                    color: "black",
                    fontWeight: "bold",
                    fontSize: 18,
                  }}
                >
                  {item.name.charAt(0)}
                </Text>
              </View>
            )}
          </TouchableOpacity>
        </View>
        <TouchableOpacity
          style={{ width: "82%" }}
          testID="listItemButton"
          onPress={() =>
            this.props.navigation.navigate("ConversationScreen", {
              chatNumber: item.uid,
              mainConversationId: item.conversationId,
              userName: item.name,
              userAvtar: item.avatar,
              userType: 1,
            })
          }
        >
          <View style={{ flexDirection: "row", alignItems: "center" }}>
            <View style={{ flexDirection: "column" }}>
              <Text
                style={{
                  fontSize: 18,
                  fontWeight: "600",
                  color: COLORS.orangelight,
                }}
              >
                {item.name}
              </Text>
              <Text
                style={{
                  marginTop: 5,
                  fontSize: 14,
                  fontWeight: "600",
                  color: COLORS.white,
                }}
              >
                Hey there! I am Using Sphara...
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );

  render() {
    return (
      <SafeAreaView style={Styles.safeareaView}>
        <View style={Styles.child}>
          <ImageBackground source={IMAGE.back1} style={Styles.container}>
            <ImageBackground source={IMAGE.back2} style={Styles.container}>
              <View
                style={{
                  padding: 20,
                  flexDirection: "row",
                }}
              >
                <TouchableOpacity
                  testID="newChatButton"
                  style={{
                    width: 10,
                    justifyContent: "center",
                    alignItems: "center",
                  }}
                  onPress={() => this.props.navigation.navigate("Chat")}
                >
                  <Image
                    source={IMAGE.image_back}
                    style={{ height: 18, width: 18, tintColor: COLORS.white }}
                  />
                </TouchableOpacity>
                <Text
                  style={{
                    color: "white",
                    width: "90%",
                    textAlign: "center",
                    fontSize: 20,
                  }}
                >
                  New Chat
                </Text>
              </View>

              {this.state.visibleTab == 0 ? (
                <>
                  <View
                    style={{
                      height: 38,
                      width: windowWidth - 25,
                      marginTop: 10,
                      alignSelf: "center",
                      flexDirection: "row",
                    }}
                  >
                    <View
                      style={{
                        width: "100%",
                        backgroundColor: COLORS.Viewback,
                        borderRadius: 100,
                        flexDirection: "row",
                        alignItems: "center",
                      }}
                    >
                      <Image
                        source={IMAGE.search}
                        style={{
                          height: 20,
                          width: 20,
                          marginLeft: 10,
                          marginRight: 5,
                          tintColor: COLORS.lightwhite,
                        }}
                      />
                      <TextInput
                        testID="searchInput"
                        placeholder="Search"
                        onChangeText={(e) => this.onSearch(e)}
                        placeholderTextColor="white"
                        style={{
                          color: "white",
                          width: "100%",
                          paddingLeft: 5,
                        }}
                      />
                    </View>
                  </View>
                  <View
                    style={{
                      width: windowWidth - 25,
                      marginTop: 15,
                      alignSelf: "center",
                      flex: 1,
                    }}
                  >
                    <FlatList
                      testID="contactList"
                      data={this.state.contactData}
                      renderItem={(item, index) => this.renderItem(item, index)}
                      keyExtractor={(item: any) => item.id}
                      showsVerticalScrollIndicator={false}
                      contentContainerStyle={{ bottom: 13 }}
                    />
                  </View>
                </>
              ) : (
                <>
                  <View>
                    <Text>Content of Private Tab</Text>
                  </View>
                </>
              )}
            </ImageBackground>
          </ImageBackground>
        </View>
      </SafeAreaView>
    );
  }
}
// Customizable Area End
