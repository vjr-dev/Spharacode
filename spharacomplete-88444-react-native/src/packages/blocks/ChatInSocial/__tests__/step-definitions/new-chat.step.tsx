//@ts-nocheck
//@ts-ignore
import { shallow, ShallowWrapper } from "enzyme";
import { defineFeature, loadFeature } from "jest-cucumber";
import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import { View } from "react-native";
import NewChat from "../../src/NewChat";

const screenProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),

    getParam: jest.fn(),
  },
  id: "Chat",
};

const feature = loadFeature("./__tests__/features/new-chat-scenario.feature");

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to New Chat", ({ given, when, then }) => {
    let newChatWrapper: ShallowWrapper;
    let instance: NewChat;
    const dummyItem = {
      item: {
        uid: "aaaaa",
        conversationId: "bbbb",
        name: "AA",
        lastMessage: "last",
        Count: 0,
        Time: "10:30",
        unReadCounts: 1,
        avatar: "aaaaa",
      },
    };
    given("I am a User loading New Chat", () => {
      newChatWrapper = shallow(<NewChat {...screenProps} />);
      expect(newChatWrapper).toBeTruthy();
      expect(newChatWrapper).toMatchSnapshot();
    });

    when("I navigate to the New Chat", () => {
      instance = newChatWrapper.instance() as NewChat;
    });

    then("I can press new chat button", () => {
      let newChatButton = newChatWrapper.findWhere(
        (node) => node.prop("testID") === "newChatButton"
      );
      newChatButton.simulate("press");
      expect(newChatButton).toBeTruthy();
    });

    then("I can enter search value in text input", async () => {
      instance.setState({ contactData: [{ name: "abcde" }] });
      let searchInput = newChatWrapper.findWhere(
        (node) => node.prop("testID") === "searchInput"
      );
      searchInput.props().onChangeText("");
      expect(searchInput).toBeTruthy();
      searchInput.props().onChangeText("abcd");
    });

    then("Display the contact list without error", () => {
      let contactList = newChatWrapper.findWhere(
        (node) => node.prop("testID") === "contactList"
      );
      contactList.props().keyExtractor({ id: 1 });
      contactList.props().renderItem(dummyItem, 0);
      expect(contactList).toBeTruthy();
    });

    then("I can press contact item", async () => {
      let newChatWrapper = shallow(
        <View>{instance.renderItem(dummyItem, 0)}</View>
      );
      let listItemButton = newChatWrapper.findWhere(
        (node) => node.prop("testID") === "listItemButton"
      );
      listItemButton.simulate("press");
      expect(listItemButton).toBeTruthy();
    });
  });
});
