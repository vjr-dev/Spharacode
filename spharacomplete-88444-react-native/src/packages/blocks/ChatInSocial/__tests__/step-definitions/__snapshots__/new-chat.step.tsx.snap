// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Chat User navigates to New Chat 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/ChatInSocial/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/ChatInSocial/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "flex": 1,
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "padding": 20,
            }
          }
        >
          <ForwardRef
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 10,
              }
            }
            testID="newChatButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/ChatInSocial/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 18,
                  "tintColor": "rgb(255,255,255)",
                  "width": 18,
                }
              }
            />
          </ForwardRef>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            New Chat
          </Text>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "height": 38,
              "marginTop": 10,
              "width": 725,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#515151",
                "borderRadius": 100,
                "flexDirection": "row",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/ChatInSocial/assets/search.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "marginLeft": 10,
                  "marginRight": 5,
                  "tintColor": "#cdcdcd",
                  "width": 20,
                }
              }
            />
            <Component
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Search"
              placeholderTextColor="white"
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "white",
                  "paddingLeft": 5,
                  "width": "100%",
                }
              }
              testID="searchInput"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
              "marginTop": 15,
              "width": 725,
            }
          }
        >
          <FlatList
            contentContainerStyle={
              Object {
                "bottom": 13,
              }
            }
            data={Array []}
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            removeClippedSubviews={true}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsVerticalScrollIndicator={false}
            testID="contactList"
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </View>
      </ImageBackground>
    </ImageBackground>
  </View>
</ForwardRef(SafeAreaView)>
`;
